src/library/searchquery.cpp

```diff
@@ -201,8 +210,13 @@ QString TextFilterNode::toSql() const {
             argument.append('_');
         }
     }
-    QString escapedArgument = escaper.escapeString(
-            kSqlLikeMatchAll + argument + kSqlLikeMatchAll);
+    QString escapedArgument;
+    if (m_matchMode == StringMatch::Equals) {
+        escapedArgument = escaper.escapeString(argument);
+    } else {
+        escapedArgument = escaper.escapeString(
+                kSqlLikeMatchAll + argument + kSqlLikeMatchAll);
+    }
```

Comment:
Less brittle and more expressive (imo) solution:
```suggestion
    QString escapedArgument;
    // use a switch-case without default case so get a compile-time -Wswitch warning
    switch (m_matchMode) {
    case StringMatch::Contains:
        escapedArgument = escaper.escapeString(
                kSqlLikeMatchAll + argument + kSqlLikeMatchAll);
    case StringMatch::Equals:
        escapedArgument = escaper.escapeString(argument);
    }
```